packages/pytest-simcore/src/pytest_simcore/httpx_calls_capture.py

```diff
@@ -0,0 +1,273 @@
+
+## Creating Mock Captures (every time you want to create/update the mock)
+- Initialize the real backend.
+- Execute tests using the command: `pytest --spy-httpx-calls-enabled=true --spy-httpx-calls-capture-path="my-captures.json"`.
+- Terminate the real backend once testing is complete.
+
+## Configuring Tests with Mock Captures (once)
+- Transfer `my-captures.json` to the `tests/mocks` directory.
+- Utilize `create_respx_mock_from_capture(..., capture_path=".../my-captures.json", ...)` to automatically generate a mock for your tests.
+
+## Utilizing Mocks (normal test runs)
+- Conduct your tests without enabling the spy, i.e., do not use the `--spy-httpx-calls-enabled` flag.
+
+
+"""
+
+import json
+from collections.abc import Callable
+from pathlib import Path
+from typing import Any, get_args
+
+import httpx
+import pytest
+import respx
+import yaml
+from pydantic import parse_obj_as
+from pytest_mock import MockerFixture, MockType
+from pytest_simcore.helpers.httpx_client_base_dev import AsyncClientCaptureWrapper
+from pytest_simcore.helpers.utils_docker import get_service_published_port
+from pytest_simcore.helpers.utils_envs import EnvVarsDict
+from pytest_simcore.helpers.utils_host import get_localhost_ip
+
+from .helpers.httpx_calls_capture_models import (
+    CreateRespxMockCallback,
+    HttpApiCallCaptureModel,
+    PathDescription,
+    SideEffectCallback,
+)
+from .helpers.httpx_calls_capture_openapi import ServiceHostNames
+
+_DEFAULT_CAPTURE_PATHNAME = "spy-httpx-calls-capture-path.json"
+
+
+def pytest_addoption(parser: pytest.Parser):
+    simcore_group = parser.getgroup("simcore")
+    simcore_group.addoption(
+        "--spy-httpx-calls-enabled",
+        action="store",
+        type=bool,
+        default=False,
+        help="If set, it activates a capture mechanism while the tests is running that can be used to generate mock data in respx",
+    )
+    simcore_group.addoption(
+        "--spy-httpx-calls-capture-path",
+        action="store",
+        type=Path,
+        default=None,
+        help=f"Path to json file to store capture calls from httpx clients during the tests. Otherwise using a temporary path named {_DEFAULT_CAPTURE_PATHNAME}",
+    )
+
+
+@pytest.fixture(scope="session")
+def spy_httpx_calls_enabled(request: pytest.FixtureRequest) -> bool:
+    return bool(request.config.getoption("--spy-httpx-calls-enabled"))
+
+
+@pytest.fixture(scope="session")
+def spy_httpx_calls_capture_path(
+    request: pytest.FixtureRequest, tmp_path_factory: pytest.TempPathFactory
+) -> Path:
+    if capture_path := request.config.getoption("--spy-httpx-calls-capture-path"):
+        assert isinstance(capture_path, Path)
+    else:
+        capture_path = (
+            tmp_path_factory.mktemp("session_fixture_spy_httpx_calls_capture_path")
+            / _DEFAULT_CAPTURE_PATHNAME
+        )
+    assert capture_path.suffix == ".json"
+    capture_path.touch()
+    return capture_path
```

Comment:
maybe it would make sense to call `capture_path.resolve()` here, to ensure the full path is used. What I worry is that the tests which are run are in subdirectories and that somehow by the time the capture json file is generated it is not placed at the expected location.